pkg/linkdiscovery/controller.go

```diff
@@ -197,6 +211,36 @@ func (c *Controller) setStateIf(condition State, state State) {
 	}
 }
 
+// FIXME: rework this function
+func (c *Controller) updateHost(macString string, ipString string, port string) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if _, ok := c.hosts[macString]; !ok {
+		host := &Host{
+			MAC:  macString,
+			IP:   ipString,
+			Port: port,
+		}
+		host.LastUpdate = time.Now()
+		c.hosts[macString] = host
+		log.Infof("Added a new host: %s <- %s/%s", macString, ipString, port)
+	}
+}
+
+// FIXME: rework this function
+func (c *Controller) pruneHosts() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	limit := time.Now().Add(-30 * time.Minute) // this is to discuss
```

Comment:
Yep, we'll probably make this configurable via gNMI.

Reply:
Alright, left unchanged per our discussion.